src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -605,16 +612,35 @@ public HttpResponse onSuccess(String authorizationCode) {
         };
     }
     
-    public void updateAuthorizationStrategy(OAuthSession oauth) throws IOException, GeneralSecurityException {
+    public UsernamePasswordAuthenticationToken updateAuthorizationStrategy(Credential credential) throws IOException, GeneralSecurityException {
     	populateDefaults();
-        OpenShiftUserInfo info = getOpenShiftUserInfo(oauth.getCredential(), transport);
-        Set<String> allowedRoles = postSAR(oauth.getCredential(), transport);
+        OpenShiftUserInfo info = getOpenShiftUserInfo(credential, transport);
+        Set<String> allowedRoles = postSAR(credential, transport);
         GrantedAuthority[] authorities = new GrantedAuthority[] {SecurityRealm.AUTHENTICATED_AUTHORITY};
+                
+        // we append the role suffix to the name stored into Jenkins, since a given user is able to log in at varying scope/permission
+        // levels in openshift; however, for now, we make sure the display name for Jenkins does not include this suffix
+        String suffix = "";
+        if (allowedRoles.contains("admin"))
+            suffix = "-admin";
+        else if (allowedRoles.contains("edit"))
+            suffix = "-edit";
+        else if (allowedRoles.contains("view"))
+            suffix = "view";
```

Comment:
"myuser.view" is never your actual username.  It is only the key into the jenkins auth matrix, which is constructed based on your actual username ("myuser") and the self-sar results of your token.  The key *can* change for the current session as the permissions for username/token changes.

This level of indirection hence avoids the race scenario.